### ISSUE TITLE
Pass ArgumentConverter to ArgumentAccessor::get

### ISSUE THREAD

**Issue by nipafx** [junit-team/junit5#1516]
TL;DR: On `ArgumentAccessor`, implement `<T> T get(int, Class<T>, ArgumentConverter)` or `<T> T get(int, ArgumentConverter<T>)`

## Current situation

Say there's a parameterized test that uses `ArgumentAccessor` and wants to convert a single argument to a custom type. This doesn't work out of the box if that type has no suitable factory:

```java
@ParameterizedTest
@CsvSource({ "0" })
void testPointNorm_(ArgumentsAccessor arguments) {
	// "ArgumentAccessException: Argument at index [0] with value [0] and
 	// type [String] could not be converted or cast to type [NoFactory].""
	NoFactory nope = arguments.get(0, NoFactory.class);
	assertNotNull(nope);
}

static class NoFactory {
	NoFactory(int __) { }
}
```

## Proposal

Would be nice to convert that `String` to a `NoFactory`. If only there was some kind of converter... oh, wait: 😉

```java
@ParameterizedTest
@CsvSource({ "0" })
void testPointNorm_(ArgumentsAccessor arguments) {
	NoFactory yay = arguments.get(0, NoFactory.class, NoFactoryConverter.class);
	assertNotNull(yay);
}
```

(Using the converter manually is near impossible because it needs a `ParameterContext`.)

Now, if `ArgumentConverter` were generic in the returned type, `get` could even elide the type token:

```java
<T> T get(int, ArgumentConverter<T>);
```

**Comment by marcphilipp** [comment]
Thanks for the proposal! However, `ArgumentsAccessor` can't just create a `ParameterContext` for a non-existing parameter either, can it? 🤔

**Comment by nicolaiparlog** [comment]
That's a pretty good and fairly obvious point... 😆

We could use the `ArgumentsAccessor arguments` parameter itself to generate a `ParameterContext`, but I can see how that would be a little shady... On the other hand, I think it would still cover a lot of use cases as conversion will often be possible just from input data alone, i.e. without even having to look at the parameter context.

Even if there is a need to use specific annotations for a conversion (something the `ArgumentCoverter` would look for on what is usually a regular parameter), it could be applied to `ArgumentsAccessor arguments` just as well.
